src/Microsoft.Sbom.Api/PackageDetails/ComponentDetailsUtils/RubyGemsUtils.cs

```diff
@@ -0,0 +1,291 @@
+            }
+            else
+            {
+                licenseField = null;
+            }
+
+            return new ParsedPackageInformation(name.First(), version.First(), new PackageDetails(licenseField, supplierField));
+        }
+        catch (PackageMetadataParsingException e)
+        {
+            log.Error("Error encountered while extracting supplier info from gemspec file. Supplier information may be incomplete.", e);
+            recorder.RecordMetadataException(e);
+
+            return null;
+        }
+    }
+
+    /// <summary>
+    /// Given a string and a regex pattern this method will return a List of strings containing all matches found.
+    /// </summary>
+    /// <param name="content">string of text to be pattern matched.</param>
+    /// <param name="pattern">regex expression to use on the content.</param>
+    /// <returns>A list of strings containing all matches found. If none are found an empty list will be returned.</returns>
+    private List<string> GetMatchesFromPattern(string content, string pattern)
+    {
+        var matches = new List<string>();
+        var match = Regex.Match(content, pattern);
+        if (match.Success)
+        {
+            var matchValue = match.Groups[1].Value;
+            var matchesArray = matchValue.Split(','); // Splitting multiple authors/licenses
+            foreach (var item in matchesArray)
+            {
+                var value = item.Trim().Trim('"'); // Remove extra spaces and quotes
+                matches.Add(value);
+            }
+        }
+
+        return matches;
+    }
+
+    /// <summary>
+    /// Runs the 'where gem' or "which gem" command based on OS and returns a single path to the gem executable.
+    /// </summary>
+    /// <returns>A path to the location of the gem executable.</returns>
+    private string FindGemExecutablePath()
+    {
+        try
+        {
+            var processStartFindGem = null as ProcessStartInfo;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                processStartFindGem = new ProcessStartInfo("where", "gem")
+                {
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    UseShellExecute = false,
+                    CreateNoWindow = true
+                };
+            }
+            else
+            {
+                processStartFindGem = new ProcessStartInfo("which", "gem")
+                {
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    UseShellExecute = false,
+                    CreateNoWindow = true
+                };
+            }
+
+            using var processFindGem = new Process();
+            processFindGem.StartInfo = processStartFindGem;
+
+            processFindGem.Start();
+            var gemExecutablePath = processFindGem.StandardOutput.ReadToEnd()?.Trim().ToString();
```

Comment:
Have you tested this locally? This code feels race-conditiony.

Reply:
Haven't had issues on either Mac or Windows. Should I add logic to wait for the process to finish + include a timeout?

Reply:
Definitely supply a timeout. If I understand correctly the standardout.ReadToEnd should only complete once the process completes (one way or another) so I don't THINK additional waiting should be needed.